ksql-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -123,6 +125,22 @@ public TopicDeleteInjector(
     return statement.withStatement(withoutDeleteText, withoutDelete);
   }
 
+  private void checkTopicError(final Exception error, final String sourceName) {
+    if (!(error instanceof KafkaDeleteTopicsException
+            && ((KafkaDeleteTopicsException) error).getExceptionList().get(0).getValue()
+            instanceof UnknownTopicOrPartitionException)) {
+      throw new KsqlException("Could not delete the corresponding kafka topic: "
```

Comment:
Don't throw `KsqlException` in this case. You can wrap in a `RuntimeException` instead.